power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -130,38 +126,37 @@ UpdateCompProperties check_component_independence(ConstDataset const& update_dat
     return properties;
 }
 
-inline void validate_update_data_independence(UpdateCompProperties const& comp) {
+template <class... ComponentTypes>
+using UpdateIndependence = std::array<UpdateCompProperties, utils::n_types<ComponentTypes...>>;
+
+inline void validate_update_data_independence(UpdateCompProperties const& comp, std::string const& comp_name) {
```

Comment:
I understand the use of `string` inside `UpdateCompProperties` is suboptimal, but this change feels like un-necessary.

Reply:
This was introduced for one very specific reason. I wanted to make `UpdateCompProperties` trivially copyable, and the only way I saw was to remove string from there, hence why it is here (and elsewhere) now. And this came from the fact that I wanted to remove all "independence" logic from the `get_sequence_idx` functions.

Reply:
Well, necessary trade-off I guess